core/lib/multivm/src/versions/vm_latest/tests/block_tip.rs

```diff
@@ -168,7 +179,15 @@ fn execute_test(test_data: L1MessengerTestData) -> u32 {
 
     let ergs_after = vm.vm.state.local_state.callstack.current.ergs_remaining;
 
-    ergs_before - ergs_after
+    assert_eq!(ergs_before - ergs_after, result.statistics.gas_used);
+
+    TestStatistics {
+        max_used_gas: ergs_before - ergs_after,
+        pubdata_published: result.statistics.pubdata_published,
```

Comment:
VM doesn't calculate `pubdata_published` for batch tip. It does it only for transactions when it calculates refunds for them. That's why you have `MAX_BATCH_TIP_PUBDATA_PUBLISHED: u64 = 0`